test/index.spec.js

```diff
@@ -86,35 +86,24 @@ describe('Pubsub Peer Discovery', () => {
     })
   })
 
-  it('should be able to encode/decode a response', async () => {
-    const discovery = new PubsubPeerDiscovery({ libp2p: mockLibp2p })
-    const id = randomBytes(32)
-    const peerId = await PeerID.create({ bits: 512 })
-    const expectedPeerInfo = new PeerInfo(peerId)
-    expectedPeerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/8082/ws')
-    expectedPeerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/8083/ws')
-
-    const queryResponse = {
-      queryID: id,
-      publicKey: peerId.pubKey.bytes,
-      addrs: expectedPeerInfo.multiaddrs.toArray().map(ma => ma.buffer)
-    }
+  it('should not broadcast if only listening', async () => {
+    discovery = new PubsubPeerDiscovery({ libp2p: mockLibp2p, listenOnly: true })
 
-    const deferred = defer()
-    const encodedResponse = PB.QueryResponse.encode(queryResponse)
-    discovery.on('peer', (p) => deferred.resolve(p))
+    sinon.spy(mockLibp2p.pubsub, 'publish')
+    await discovery.start()
+    expect(mockLibp2p.pubsub.publish.callCount).to.equal(0)
```

Comment:
We should add a delay here. Also, `discovery.start()` does not return a promise

Reply:
We actually don't need to. I removed the await from start. All the broadcast code is synchronous so we're guaranteed publish will either be called or it won.t